src/pages/index.tsx

```diff
@@ -21,16 +31,103 @@ import tClientNext from "../tClientNext";
 import PublicUser from '../shared/publicModels/PublicUser';
 import PublicGroup from '../shared/publicModels/PublicGroup';
 import { MdVideocam } from 'react-icons/md';
+import { HSeparator } from 'horizon-ui/components/separator/Separator';
+import { IYuanjianUser } from 'shared/user';
+import { toast } from "react-toastify";
 
 const AppIndex: NextPageWithLayout = () => {
   const user = useUserInfo();
+  if(!user.name){
+    return <Box paddingTop={'80px'}><SetName u = {user} /><Meetings /></Box>
+  };
   return <Box paddingTop={'80px'}><Meetings /></Box>;
 }
 
 AppIndex.getLayout = (page) => <AppLayout>{page}</AppLayout>;
 
 export default AppIndex;
 
+function SetName({u} : {u:IYuanjianUser}) {
+  const textColor = useColorModeValue('navy.700', 'white');
+  const brandStars = useColorModeValue('brand.500', 'brand.400');
+  const [isOpen, setOpen] = useState(true);
+  const [name, setName] = useState('');
+  
+  const handleSubmit = async () => {
+    if (name) {
+      const updatedUser: IYuanjianUser = {
+        id: u.id,
+        pinyin: name,
+        name: name,
+        email: u.email,
+        roles: u.roles,
+        clientId: u.clientId,
```

Comment:
why do we need to pass in these fields even if they are not updated?

Reply:
the return object was IYuanjianUser type. The type is removed and an object with only updated name is used now